packages/teams-js/test/public/clipboard.spec.ts

```diff
@@ -0,0 +1,480 @@
+import { errorLibraryNotInitialized } from '../../src/internal/constants';
+import { GlobalVars } from '../../src/internal/globalVars';
+import { DOMMessageEvent } from '../../src/internal/interfaces';
+import { clipboard } from '../../src/public';
+import { app } from '../../src/public/app';
+import { errorNotSupportedOnPlatform, FrameContexts, HostClientType } from '../../src/public/constants';
+import { Utils } from '../utils';
+
+/* eslint-disable */
+/* As part of enabling eslint on test files, we need to disable eslint checking on the specific files with
+   large numbers of errors. Then, over time, we can fix the errors and reenable eslint on a per file basis. */
+
+describe('clipboard', () => {
+  const dataToCopy: Blob = new Blob(['Mock data'], { type: 'text/plain' });
+  const htmlDataToCopy: Blob = new Blob(['<p>Mock data</p>'], { type: 'text/html' });
+  const imageJpegDataToCopy: Blob = new Blob(['Mock data'], { type: 'image/jpeg' });
+  const imagePngdataToCopy: Blob = new Blob(['Mock data'], { type: 'image/png' });
+  const base64ResponsePng = JSON.stringify({ mimeType: 'image/png', content: 'SGVsbG8gd29ybGQ' });
+  const base64ResponseJpeg = JSON.stringify({ mimeType: 'image/jpeg', content: 'SGVsbG8gd29ybGQ' });
+  const base64ResponsePlainText = JSON.stringify({ mimeType: 'text/plain', content: 'SGVsbG8gd29ybGQ' });
+  const base64ResponseHtmlText = JSON.stringify({ mimeType: 'text/html', content: 'SGVsbG8gd29ybGQ' });
+  const dataToCopyNotSupported: Blob = new Blob(['Mock data not supported'], { type: 'application/json' });
+  const allowedContexts = [FrameContexts.content, FrameContexts.task, FrameContexts.stage, FrameContexts.sidePanel];
+  Object.assign(navigator, {
+    clipboard: {},
+  });
+
+  describe('Testing clipboard.write function', () => {
+    describe('frameless', () => {
+      let utils: Utils = new Utils();
+
+      beforeEach(() => {
+        utils = new Utils();
+        utils.mockWindow.parent = undefined;
+        utils.messages = [];
+        GlobalVars.isFramelessWindow = false;
+      });
+
+      afterEach(() => {
+        app._uninitialize();
+        GlobalVars.isFramelessWindow = false;
+      });
+
+      it('clipboard.write should not allow calls before initialization', async () => {
```

Comment:
would be good to have test cases around missing or incomplete Blob passed to clipboard.write (mimeType is empty, blob is null, blob content is null or empty, etc.)

Reply:
Updated in the latest commit